README.md

```diff
@@ -73,12 +73,62 @@ const collectionAssetsResponse = await client.listAssets({
 });
 ```
 
+### Generating Stark (Layer 2) keys
+
+Stark keys are required by users in order to transact on ImmutableX's StarkEx Layer 2. They are the equivalent of Ethereum keys on L1, and allows users to sign transactions, and receive and store tokens.
+
+#### Key registration
+
+On ImmutableX, the goal of generating a Stark key is to [register](https://docs.x.immutable.com/docs/how-to-register-users/) a mapping between this key and the user's Ethereum key so that transactions requiring both L1 and L2 signers can be executed by users.
+
+#### How to generate Stark keys on ImmutableX
+
+ImmutableX provides two Stark key generation methods:
+| Type of Stark key generated | How do applications connect to these Stark accounts? | Factors determining when this method should be used: | ImmutableX tools: |
```

Comment:
Consider making this

| Type of Stark key generated: | User Connection Method: | When to use this method: | ImmutableX tools: |